src/Components/Rooms/Rooms.tsx

```diff
@@ -32,62 +38,133 @@ const StyledTableRow = styled(TableRow)(({ theme }) => ({
   },
 }));
 
-export default function Rooms() {
-const [age, setAge] =useState('');
-const handleChange = (event: SelectChangeEvent) => {
-  setAge(event.target.value);
+const style = {
+  position: 'absolute' as 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  borderRadius: "7px",
+
+  boxShadow: 24,
+  p: 4,
 };
-const{rooms,getAllRooms}=useRooms();
-const navigate=useNavigate()
-useEffect(()=>{getAllRooms()},[])
+export default function Rooms() {
+  
+    
+    const{rooms,getAllRooms}=useRooms();
+    const navigate=useNavigate()
+  const [age, setAge] = useState('');
+  const [modalState, setModalState] = useState('close')
+  const [roomId, setRoomId] = useState(0)
+  const handleChange = (event: SelectChangeEvent) => {
+    setAge(event.target.value);
+  };
+  let reqHeaders = 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2NThhMTgyYjQ3ZWUyYjE0Zjk1NDY5OTAiLCJyb2xlIjoiYWRtaW4iLCJ2ZXJpZmllZCI6ZmFsc2UsImlhdCI6MTcwNDQ4NDEyNiwiZXhwIjoxNzA1NjkzNzI2fQ.N9gU4yHP3g8g5ajsm_Tf6w1EIDJE-Gfu4e0tsPejUj8'
+  let Headers = { Authorization: reqHeaders }
+
+
+
+  const showDeleteModal = (id) => {
+    setModalState('delete-modal')
+    setRoomId(id)
+
+  }
+  const handleClose = () => setModalState("close");
+
+  const deleteRoom = () => {
+    axios.delete(`http://154.41.228.234:3000/api/v0/admin/rooms/${roomId}`, { headers: Headers })
+      .then((response) => {
+        console.log(response);
+        setRoomId(roomId);
+        handleClose();
+        getAllRooms()
+
+      }).catch((error) => {
+        console.log(error);
+      })
+  }
+  useEffect(() => { getAllRooms() }, [])
```

Comment:
We can remove this effect and use the rooms directly as we already run this effect in the custom hook.

Reply:
You need also to move the getAllRooms to the effect int the useRooms file as we only need it inside the effect. This will remove the eslint warning.
![image](https://github.com/FatmaHossam5/stayCation_Booking/assets/64327685/69dcb812-3b29-4dc9-8335-14ba86e7a036)